php/utils.php

```diff
@@ -1505,3 +1504,72 @@ function format_table( $items ) {
 	}
 	\EE::log( $delem );
 }
+
+/**
+ * Get the site-name from the path from where ee is running if it is a valid site path.
+ *
+ * @return bool|String Name of the site or false in failure.
+ */
+function get_site_name() {
+	$sites = EE::db()::select( array( 'sitename' ) );
+
+	if ( $sites ) {
+		$cwd          = getcwd();
+		$name_in_path = explode( '/', $cwd );
+		$site_name    = array_intersect( array_flatten( $sites ), $name_in_path );
```

Comment:
I think it would make more sense if you change `$site_name` to `$sites` (since there could be multiple site name matching in DB).